### PR TITLE
Add unsupported version checks for major 3.11 and 3.12 features

### LINKED ISSUE #9791: Add check for using major 3.11 and 3.12 features with lower `--py-version`
### Current problem

Add `using-exception-group-in-unsupported-version` and
`using-generic-type-syntax-in-unsupported-version` for uses of Python 3.11+ or
3.12+ features on lower supported versions provided with `--py-version`.

### Desired solution

new msgs

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :sparkles: New feature |

## Description
Closes #9791

### CODE REVIEW

File: pylint/checkers/unsupported_version.py [hunk 1]
```diff
@@ -79,6 +93,51 @@ def _check_typing_final(self, node: nodes.Decorators) -> None:
                 "using-final-decorator-in-unsupported-version", node=decorator
             )
 
+    @only_required_for_messages("using-exception-groups-in-unsupported-version")
+    def visit_trystar(self, node: nodes.TryStar) -> None:
+        if not self._py311_plus:
+            self.add_message("using-exception-groups-in-unsupported-version", node=node)
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/unsupported_version.py:99]: I'm not sure we can even visit a trystar if we're not in python 3.11+ (the ast for 3.10 does not have the construct, we had to add it at this point, right ?)

**Review by Pierre-Sassoulas — COMMENTED** [review]
Great idea !

File: pylint/checkers/unsupported_version.py (hunk 1 above)
Comment (jacobtylerwalls) [reply pylint/checkers/unsupported_version.py:99]: Right. But we need to know whether the lowest supported version is 3.11 or lower.

File: doc/data/messages/a/anomalous-backslash-in-string/bad.py [hunk 2]
```diff
@@ -1 +1 @@
-string = "\z"  # [anomalous-backslash-in-string]
+string = "\z"  # [syntax-error]
```
Comment (DanielNoord) [inline doc/data/messages/a/anomalous-backslash-in-string/bad.py:1]: We should probably move these then? As this will look weird in our documentation.

**Review by DanielNoord — APPROVED** [review]
Do  we want functional tests for these?

File: doc/data/messages/a/anomalous-backslash-in-string/bad.py (hunk 2 above)
Comment (jacobtylerwalls) [reply doc/data/messages/a/anomalous-backslash-in-string/bad.py:1]: I'm not sure what the best solution is.

Comment (DanielNoord) [reply doc/data/messages/a/anomalous-backslash-in-string/bad.py:1]: Perhaps add a `details.rst` explaining that this is because of our CI?

**Review by Pierre-Sassoulas — COMMENTED** [review]
I think we should cover the new code with tests. Regarding the doc not raising the right message anymore, maybe we need run the CI for each supported interpreters and add skipif on the one that are not relevant (with the same configuration than for functional test min_pyver / max_pyver) ?

**Comment by jacobtylerwalls** [comment]
Will come back and add some tests.

> Regarding the doc not raising the right message anymore, maybe we need run the CI for each supported interpreters and add skipif on the one that are not relevant (with the same configuration than for functional test min_pyver / max_pyver) ?

I wouldn't object, but I'm not volunteering to work on it in the near term.

**Review by DanielNoord — DISMISSED** [review]
I guess we're still missing some tests but I would be okay with merging already. This seems like a useful new feature.

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**